sandbox/apparmor/apparmor.go

```diff
@@ -170,6 +170,40 @@ func ParserMtime() int64 {
 	return mtime
 }
 
+// PromptingSupported returns true if prompting is supported by the system.
+// Otherwise, returns false, along with a string explaining why prompting is
+// unsupported.
+func PromptingSupported() (bool, string) {
+	kernelFeatures, err := appArmorAssessment.KernelFeatures()
+	if err != nil {
+		return false, fmt.Sprintf("cannot check apparmor kernel features: %v", err)
+	}
+	if !KernelFeaturesSupportPrompting(kernelFeatures) {
+		return false, "apparmor kernel features do not support prompting"
+	}
+	parserFeatures, err := appArmorAssessment.ParserFeatures()
+	if err != nil {
+		return false, fmt.Sprintf("cannot check apparmor parser features: %v", err)
+	}
+	if !ParserFeaturesSupportPrompting(parserFeatures) {
+		return false, "apparmor parser does not support the prompt qualifier"
+	}
+	// TODO: return true once snapd supports prompting
+	return false, "requires newer version of snapd"
```

Comment:
we have a disconnect between the hard coded false here and the result of  KernelFeaturesSupportPrompting&& ParserFeaturesSupportPrompting which I don't think we want because we would be back to snapd and snap run disagreeing on things.  We might want to consolidate  KernelFeaturesSupportPrompting and ParserFeaturesSupportPrompting into:

`PromptingSupportedGivenFeatures(features Features) (bool, string)`

where Features is some kind of struct with Kernel and Parser fields,  then the default to false can happen in this function which is also called by no-arguments PromptingSupported and results are consistent